app/page.tsx

```diff
@@ -86,6 +88,17 @@ function ControlBar(props: {
   onConnectButtonClicked: () => void;
   agentState: AgentState;
 }) {
+  // Add Krisp noise filter to the users microphone track.
+  const { microphoneTrack } = useLocalParticipant();
+  const krisp = useKrispNoiseFilter();
+  useEffect(() => {
+    if (microphoneTrack && krisp?.processor) {
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-expect-error
+      microphoneTrack.track?.setProcessor(krisp.processor);
+    }
+  }, [krisp?.processor, microphoneTrack]);
+
```

Comment:
`krisp.setNoiseFilterEnabled(true)` is enough, manually setting the processor is not requried, neither is the `useLocalParticipant` hoook